File: apps/mobile-android/app/src/main/kotlin/com/maimai/home/data/models/EventEnvelope.kt
package com.maimai.home.data.models

import androidx.annotation.Keep
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

@Keep
@Serializable
data class EventEnvelope(
    val type: String,
    val payload: JsonElement,
    val timestamp: String,
)

File: services/windows-agent/src/MaimaiHomeAgent/Realtime/EventEnvelope.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace MaimaiHomeAgent.Realtime;

/// <summary>
///     Wire envelope sent over the /api/events WebSocket. Field names use lower-camel
///     when serialized so mobile/web clients can parse the payload directly.
/// </summary>
public sealed record EventEnvelope(
    [property: JsonPropertyName("type")] string Type,
    [property: JsonPropertyName("payload")]
    JsonElement Payload,
    [property: JsonPropertyName("timestamp")]
    DateTimeOffset Timestamp);